### PR TITLE
Add long double format option to h5dump

### PR DESCRIPTION
**Pull request by byrnHDF**
In trying to fix tfloatattrs, discovered that the floating point format option did not apply to long double datatypes.
This adds the missing option.

### CODE REVIEW

**Comment by derobins** [comment]
Needs a release note!

**Review by lrknox — APPROVED** [review]

**Review by derobins — APPROVED** [review]

**Review by mattjala — APPROVED** [review]

File: tools/test/h5dump/expected/tfloatsattrs.ddl
Comment (jhendersonHDF) [inline tools/test/h5dump/expected/tfloatsattrs.ddl:1]: @byrnHDF Was this file updated correctly? It looks like the `--lformat` wasn't applied; the data still has varying numbers of digits after the decimal

**Comment by byrnHDF** [comment]
Yes, but I used %.4Lg and not %.04Lg

________________________________________

**Comment by jhendersonHDF** [comment]
> Yes, but I used %.4Lg and not %.04Lg

`%.4Lg` is the correct specifier; it just looks like it wasn't applied

**Pull request merged**